src/__tests__/token-registry.e2e.test.ts

```diff
@@ -0,0 +1,295 @@
+      return transaction;
+    };
+
+    it("should be able to change title escrow holder", async () => {
+      const tokenId = "0x0000000000000000000000000000000000000000000000000000000000000004";
+      const retrievedTokenRegistryAddress = await getTokenRegistryAddress();
+      await mintTransaction(retrievedTokenRegistryAddress, tokenId);
+      const transaction = await changeHolder(retrievedTokenRegistryAddress, tokenId);
+      expect(transaction.confirmations).toBeGreaterThanOrEqual(1);
+      expect(transaction.status).toBe(1);
+    });
+
+    it("should be able to change title escrow owner", async () => {
+      const tokenId = "0x0000000000000000000000000000000000000000000000000000000000000005";
+      const retrievedTokenRegistryAddress = await getTokenRegistryAddress();
+      await mintTransaction(retrievedTokenRegistryAddress, tokenId);
+      const transaction = await changeOwner(retrievedTokenRegistryAddress, tokenId);
+      expect(transaction.confirmations).toBeGreaterThanOrEqual(1);
+      expect(transaction.status).toBe(1);
+    });
+
+    it("should be able to nominate title escrow beneficiaries", async () => {
+      const tokenId = "0x0000000000000000000000000000000000000000000000000000000000000006";
+      const retrievedTokenRegistryAddress = await getTokenRegistryAddress();
+      await mintTransaction(retrievedTokenRegistryAddress, tokenId);
+      const transaction = await nominateTitleEscrowBeneficiary(retrievedTokenRegistryAddress, tokenId);
+
+      expect(transaction.confirmations).toBeGreaterThanOrEqual(1);
+      expect(transaction.status).toBe(1);
+    });
+
+    it("should be able to complete transfer of title escrow beneficiaries", async () => {
+      const tokenId = "0x0000000000000000000000000000000000000000000000000000000000000007";
+      const retrievedTokenRegistryAddress = await getTokenRegistryAddress();
+      await mintTransaction(retrievedTokenRegistryAddress, tokenId);
+      await nominateTitleEscrowBeneficiary(retrievedTokenRegistryAddress, tokenId);
+      const transactionParameter: TitleEscrowNominateBeneficiaryCommand = {
+        tokenRegistry: retrievedTokenRegistryAddress,
+        tokenId: tokenId,
+        newOwner: accounts.receiver.ethAddress,
+        ...defaults,
+      };
+      const transaction = await endorseNominatedBeneficiary(transactionParameter);
+
+      expect(transaction.nominatedBeneficiary).toBe(accounts.receiver.ethAddress);
+      expect(transaction.transactionReceipt.confirmations).toBeGreaterThanOrEqual(1);
+      expect(transaction.transactionReceipt.status).toBe(1);
+    });
+  });
+
+  const getTokenRegistryAddress = async (): Promise<string> => {
+    if (tokenRegistryAddress === "") {
+      const delay = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+      await delay(3000);
+      console.log("Delayed By 3s");
+      return await getTokenRegistryAddress();
+    } else {
```

Comment:
this will be recursive and stuck forever no? if bad case of `tokenRegistryAddress === ""` happens